src/Microsoft.Extensions.FileProviders.Physical/PollingWildCardChangeToken.cs

```diff
@@ -153,16 +154,26 @@ private static bool ArrayEquals(byte[] previousHash, byte[] currentHash)
             return true;
         }
 
-        private void ComputeHash(IncrementalHash sha256, string path)
+        private void ComputeHash(IncrementalHash sha256, string value, DateTime lastChangedUtc)
         {
-            var byteCount = Encoding.Unicode.GetByteCount(path);
-            if (_fileNameBuffer == null || byteCount > _fileNameBuffer.Length)
+            var byteCount = Encoding.Unicode.GetByteCount(value);
+            if (_byteBuffer == null || byteCount > _byteBuffer.Length)
             {
-                _fileNameBuffer = new byte[Math.Max(byteCount, 256)];
+                _byteBuffer = new byte[Math.Max(byteCount, 256)];
             }
 
-            var length = Encoding.Unicode.GetBytes(path, 0, path.Length, _fileNameBuffer, 0);
-            sha256.AppendData(_fileNameBuffer, 0, length);
+            var length = Encoding.Unicode.GetBytes(value, 0, value.Length, _byteBuffer, 0);
+            sha256.AppendData(_byteBuffer, 0, length);
+
+            Debug.Assert(_byteBuffer.Length > sizeof(long));
+            unsafe
+            {
+                fixed (byte* b = _byteBuffer)
+                {
+                    *((long*)b) = lastChangedUtc.Ticks;
```

Comment:
Do we typically do unsafe code to avoid all the bit masking that would otherwise be required here?